### ISSUE TITLE
Feedback: 'Serialization'

### ISSUE THREAD

**Issue by adriatic** [Particular/docs.particular.net#3940]
Feedback for 'Serialization' https://docs.particular.net/nservicebus/serialization/

Location in GitHub: https://github.com/Particular/docs.particular.net/blob/master/nservicebus/serialization/index.md

ping @Particular/nservicebus-maintainers

After spending last 3 years in the "Node world", I might be biased, but I will take the chance to write the following comment:

After reading the section:
> The Newtonsoft JSON Serializer provides an effective general-purpose serializer appropriate for most use cases based on the ubiquitous Json.NET package. The Newtonsoft package provides a good combination of compactness, human readability, and performance. Other serializers are supported in order to enable specific performance or integration requirements.

the next statement

> The default serializer used in NServiceBus projects is the custom XmlSerializer.

begs for the question **WHY??** 

I cringe having to deal with XML after all wonderful time with Newtonsoft JSON. This ought to be the consequence of using Microsoft's MSMQ - but you could have set the default differently, I think

**Comment by SzymonPobiega** [comment]
Thanks for the feedback @adriatic. We are generally moving away from defaults in NServiceBus (e.g. in V7 MSMQ is no longer a default transport) in favour of explicit configuration. It is likely that in the next versions we'll do the same with regards to the serializer. Most people have strong opinions in this subject and there seems to be no safe default option.

**Comment by adriatic** [comment]
It is seldom the case that having a default option makes the choice easier - not having the default forces everyone to think about the available alternatives. It would probably be the best to allow setting the default version, as this would increased the developer's confidence that such setting would be respected everywhere.

**Comment by danielmarbach** [comment]
We decided to not move away from the default XML serializer. In hindsight that was a good move. The JSON space has been split into the advanced serializer with Newtonsoft as well as the fast/low allocation serializer provided by the .NET Core. But it is still an explicit choice that a user has to make. So therefore having a default XmlSerializer that supports all the bells and whistles as part of NServiceBus is totally fine to get started and after that users have to make a choice anyway. Once they do that they have to opt-in for a specific one. Closing this
